File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/SpikeSlime_L_Model.ts
import {MonsterInfo} from "mods/ModTheSpire/Scripts/Unit/Monster/MonsterInfo";
import {ILoader} from "mods/ModTheSpire/Scripts/Loader/ILoader";
import {SpineLoader} from "mods/ModTheSpire/Scripts/Loader/SpineLoader";
import FileHelper from "mods/ModTheSpire/Scripts/FileHelper";
import DungeonManager from "mods/ModTheSpire/Scripts/DungeonManager";
import {Mathf} from "Core/Module/Math/Mathf";

export class SpikeSlime_L_Model extends MonsterInfo{
    public static Id = "SpikeSlime_L"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(67, 73);
        else
            return Mathf.RandomRange(64, 70);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/slimeAltL/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/slimeAltL/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/slimeAltL/skeleton.png"));
    }

    get Name(): string {
        return "尖刺史莱姆（大）";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/SpikeSlime_L_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class SpikeSlime_L_Model : MonsterInfo
    {
        public static string Id = "SpikeSlime_L";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(67, 73) : Random.Range(64, 70);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/slimeAltL/skeleton.atlas",
                    "Monsters/theBottom/slimeAltL/skeleton.json", "Monsters/theBottom/slimeAltL/skeleton.png");
            }
        }

        public override string Name => "尖刺史莱姆（大）";

        public SpikeSlime_L_Model(Vector2 pos) : base(pos)
        {
        }
    }
}